### DISCUSSION TITLE
Any documents or guideline for new custom hardware support

### DISCUSSION THREAD

**Discussion by aaronisme** [Moddable-OpenSource/moddable#1054] [Q&A]
Hi, all 

I am a javascript lover and am also really interested in embedded devices. Really thanks for Moddable SDK which make Javascript be used on embedded devices.  Currently, I am building some projects which use an ARM Cortex-M4 + FPU microcontroller. we are using the gcc for the project firmware compilation and building. Am I wondering if is possible to add new hardware support on Moddable SDK?  

If I would like to add the custom hardware is there any documents or guideline to help?

Thanks a lot

**Comment by aaronisme** [comment]
@pmcneil @mkellner would you please provide some help on this ?

**Comment by mkellner** [comment]
Hello @aaronisme.

There was a similar discussion: #616.

There was some discussion during the porting of the Raspberry Pi Pico. There is a Discussion category "Porting" which may give some insight.

**Reply by aaronisme** [reply]
Hi @mkellner I will check the discussion you mentioned :)
